### PR TITLE
Add support for custom local-tmp-dir in LocalTarget

### PR DESCRIPTION
**Pull request by hellais**
This requires setting a client.cfg option:
[core]
local-tmp-dir: /media/ephemeral

### CODE REVIEW

File: luigi/file.py [hunk 1]
```diff
@@ -74,6 +75,9 @@ def __init__(self, path=None, format=None, is_tmp=False):
         if format is None:
             format = get_default_format()
 
+        tmp_dir = luigi.configuration.get_config().get("core", "local-tmp-dir", None)
```
Comment (Tarrasch) [inline luigi/file.py]: Don't use `core` for this and also use the new configuration system.

**Comment by Tarrasch** [comment]
This looks good, but please fix the comment and add a test as well. :)

File: luigi/file.py (hunk 1 above)
Comment (hellais) [inline luigi/file.py]: What section should it go under if not core?

Which new configuration system?

Comment (erikbern) [inline luigi/file.py]: you can define a `luigi.Config` class – see examples in the existing code

**Comment by themalkolm** [comment]
Why don't you just provide it via TMPDIR env variable?

**Comment by Tarrasch** [comment]
Can you please squash and (maybe) rebase this? I'll be very happy to merge this then!

**Comment by Tarrasch** [comment]
Reopen ticket when you have time to iterate on this please. :)

**Pull request closed without merging**